src/main.rs

```diff
@@ -2,23 +2,39 @@ use anyhow::{anyhow, Context, Result};
 use clap::ArgMatches;
 use directories::ProjectDirs;
 use rand::seq::IteratorRandom;
-use std::path::PathBuf;
+use std::{path::PathBuf, process::Command};
 
 pub mod clash;
 pub mod formatter;
 pub mod outputstyle;
-pub mod tester;
+pub mod solution;
 
 use clash::Clash;
 use outputstyle::OutputStyle;
-use tester::TestRunResult;
+use solution::Solution;
 
 #[derive(Clone)]
 pub enum OutputStyleOption {
     Default,
     Plain
 }
 
+fn command_from_argument(cmd_arg: Option<&String>) -> Result<Option<Command>> {
+    if cmd_arg.is_none() { return Ok(None) };
+
+    match shlex::split(&cmd_arg.unwrap()) {
+        Some(shlexed_cmd) if shlexed_cmd.is_empty() => Ok(None),
```

Comment:
Shouldn't this be an error? This branch is hit if you pass in `""` as `cmd_arg`. With that change the return type of the function could also be simplified to just `Result<Command>`.

Reply:
Because this function is generalized for both `--command` (required) and `--build-command` (optional), it returns an `Option<Command>` (as is the convention in our project when something is optional).

An empty `--command` errors out with an `expect` at the place where this function is called.

Reply:
To clarify the return value is `Result<Option<Command>>` because this function can:
- Be passed a missing argument which is `Ok(None)` (since this simply means something is absent, no error has occurred)
- Be passed a blank argument which is again `Ok(None)` for the same reason
- Be passed a malformed command which is an `Err`
- Be passed a correctly formatted command which is an `Ok(Some(Command))`

see rust is easy

Reply:
I think you're right about the return value type. I would still expect an empty string to be an error instead of `None` (since command always needs some executable to run) but I can see the argument that empty string is just a no-op command which is equivalent to `None`.

Reply:
I can imagine someone who is calling this executable from a script getting annoyed at getting an error with
```bash
run="ruby sol.rb"
# much later
clash run --command "$run" --build_command "$build"
```